### DISCUSSION TITLE
Issues with C language parser

### DISCUSSION THREAD

**Discussion by AlexHubner** [GumTreeDiff/gumtree#307] [Q&A]
Hello, I followed all the steps to make the gumtree tool run on my Ubuntu Linux and after many problems with Java, which were resolved, I started having problems with cgum. I couldn't do analysis with C language files even after generating the Cgum through make. The following message is displayed:

ERROR spark.http.matching.GeneralError -
java.io.IOException: Cannot run program "cgum" (in directory "/tmp"): error=2, No such file or directory

Can someone help me?

**Comment by jrfaller** [comment]
Hi!

You need to install cgum https://github.com/GumTreeDiff/cgum 

However, cgum has a lot of issues so maybe using tree-sitter C backend or srcML c backend would be a better idea?

Cheers.

**Comment by antilneeraj** [comment]
It seems like the error message is indicating that the "cgum" command is not found in your system's path, which is causing the IOException when trying to run the command.

Here are a few steps you can try to resolve the issue:

1. Check if the cgum command is installed on your system: 
   - Run `which cgum` command in the terminal to check if the command is installed and in your system's path. If the command is not found, then you need to make sure that you have built the Cgum tool successfully and installed it properly.

2. Add the Cgum executable to your system's path:
   - If the `cgum` command is not found, then you may need to add the directory containing the `cgum` executable to your system's PATH environment variable. You can do this by adding the following line to your `~/.bashrc` file:
     ```
     export PATH="/path/to/cgum/bin:$PATH"
     ```
     where "/path/to/cgum/bin" is the path to the directory containing the `cgum` executable.

3. Ensure that you have proper permissions:
   - Make sure that you have proper permissions to run the `cgum` command. You can do this by running `chmod +x cgum` in the directory where `cgum` is located.

4. Restart your terminal session:
   - If you have made changes to your system's PATH environment variable, you may need to restart your terminal session for the changes to take effect.

Once you have completed these steps, try running the analysis again and see if the issue is resolved.
